pennylane/templates/repeat.py

```diff
@@ -0,0 +1,206 @@
+        >>> ansatz()
+        >>> print(ansatz.draw())
+        1: ──H──╭C──X──H──╭C──X──H──╭C──X──┤ ⟨Z⟩
+        2: ─────╰X────────╰X────────╰X─────┤ ⟨Z⟩
+
+        **Dynamic Arguments**
+
+        In addition to passing static arguments to ``circuit``, we can also pass *dynamic* arguments.
+        These are arguments that do change with each repetition of the circuit. These parameters are passed
+        as non-keyword arguments to ``qml.repeat``, after ``circuit`` and ``depth``. Each dynamic parameter must
+        be a list of length equal to ``depth``, where each entry corresponds to the value of the argument used for the
+        corresponding layer.
+
+        For example, let us define the following variational circuit:
+
+        .. code-block:: python3
+
+            def circuit(params):
+                qml.RX(params[0], wires=[0])
+                qml.MultiRZ(params[1], wires=[0, 1])
+                qml.RY(params[2], wires=[1])
+
+        We wish to repeat this circuit 3 times, with each layer having different ``params``:
+
+        .. code-block:: python3
+
+            @qml.qnode(dev)
+            def ansatz(params):
+                qml.repeat(circuit, 3, params)
+                return [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))]
+
+        Since each application of ``circuit`` requires 3 variational parameters, and the circuit is
+        repeated 3 times, we supply an array of size (3, 3) as an argument to ``qml.repeat``:
+
+        .. code-block:: python3
+
+            params = np.array([[0.5, 0.5, 0.5], [0.4, 0.4, 0.4], [0.3, 0.3, 0.3]])
+
+        which yields the following circuit:
+
+        >>> ansatz(params)
+        >>> print(ansatz.draw())
+        0: ──RX(0.5)──╭RZ(0.5)──RX(0.4)──╭RZ(0.4)──RX(0.3)──╭RZ(0.3)───────────┤ ⟨Z⟩
+        1: ───────────╰RZ(0.5)──RY(0.5)──╰RZ(0.4)──RY(0.4)──╰RZ(0.3)──RY(0.3)──┤ ⟨Z⟩
+
+        **Passing Multiple Static and Dynamic Arguments**
+
+        It is also possible to pass multiple static and dynamic arguments into the same circuit. Dynamic
+        arguments must be ordered in ``qml.repeat`` in the same order in which they are passed into the
+        ``circuit``.
+
+        Consider the following circuit:
+
+        .. code-block:: python3
+
+            def circuit(param1, param2, wires, var):
+                qml.RX(param1, wires=wires[0])
+                qml.MultiRZ(param2, wires=wires)
+
+                if var:
+                    qml.Hadamard(wires=wires[1])
+
+        This circuit can be repeated as:
+
+        .. code-block:: python3
+
+            @qml.qnode(dev)
+            def ansatz(param1, param2):
+                qml.repeat(circuit, 2, param1, param2, wires=[1, 2], var=True)
+                return [qml.expval(qml.PauliZ(1)), qml.expval(qml.PauliZ(2))]
+
+        We can then run the circuit with a given set of parameters (note that the parameters are
+        of size (2, 1), as the circuit is repeated twice, and for each repetition, both ``param1`` and
```

Comment:
After reading up to this point, I'm still a bit confused about the shape of the dynamic parameters. This should be clearer in the documentation.

Reply:
To me it is pretty clear...you need enough params for every layer, so the first dimension must be the same as depth?

Reply:
Yep, it reminds me of the batch dimension in ML.